airbyte-workers/src/test/java/io/airbyte/workers/internal/EmptyAirbyteSourceTest.java

```diff
@@ -0,0 +1,263 @@
+    Assertions.assertThat(stateMessage.getStateType()).isEqualTo(AirbyteStateType.GLOBAL);
+    Assertions.assertThat(stateMessage.getGlobal().getSharedState()).isNull();
+    Assertions.assertThat(stateMessage.getGlobal().getStreamStates())
+        .map(streamState -> streamState.getStreamDescriptor())
+        .containsExactlyElementsOf(streamDescriptors);
+    Assertions.assertThat(stateMessage.getGlobal().getStreamStates())
+        .map(streamState -> streamState.getStreamState())
+        .containsOnlyNulls();
+
+    Assertions.assertThat(emptyAirbyteSource.attemptRead())
+        .isEmpty();
+  }
+
+  @Test
+  public void testGlobalPartial() throws Exception {
+    final String NOT_RESET_STREAM_NAME = "c";
+
+    List<StreamDescriptor> streamDescriptors = Lists.newArrayList(
+        new StreamDescriptor().withName("a"),
+        new StreamDescriptor().withName("b"),
+        new StreamDescriptor().withName(NOT_RESET_STREAM_NAME)
+    );
+
+    List<io.airbyte.config.StreamDescriptor> streamToReset = Lists.newArrayList(
+        new io.airbyte.config.StreamDescriptor().withName("a"),
+        new io.airbyte.config.StreamDescriptor().withName("b")
+    );
+
+    ResetSourceConfiguration resetSourceConfiguration = new ResetSourceConfiguration()
+        .withStreamsToReset(streamToReset);
+    WorkerSourceConfig workerSourceConfig = new WorkerSourceConfig()
+        .withSourceConnectionConfiguration(Jsons.jsonNode(resetSourceConfiguration))
+        .withState(new State()
+            .withState(Jsons.jsonNode(createGlobalState(streamDescriptors, Jsons.emptyObject())))
+        );
+
+    emptyAirbyteSource.start(workerSourceConfig, null);
+
+    Optional<AirbyteMessage> maybeMessage = emptyAirbyteSource.attemptRead();
+    Assertions.assertThat(maybeMessage)
+        .isNotEmpty();
+
+    AirbyteMessage message = maybeMessage.get();
+    Assertions.assertThat(message.getType()).isEqualTo(Type.STATE);
+
+    AirbyteStateMessage stateMessage = message.getState();
+
+    Assertions.assertThat(stateMessage.getStateType()).isEqualTo(AirbyteStateType.GLOBAL);
+    Assertions.assertThat(stateMessage.getGlobal().getSharedState()).isEqualTo(Jsons.emptyObject());
+    Assertions.assertThat(stateMessage.getGlobal().getStreamStates())
+        .filteredOn(streamState -> streamState.getStreamDescriptor().getName() != NOT_RESET_STREAM_NAME)
+        .map(AirbyteStreamState::getStreamState)
+        .containsOnlyNulls();
+    Assertions.assertThat(stateMessage.getGlobal().getStreamStates())
+        .filteredOn(streamState -> streamState.getStreamDescriptor().getName() == NOT_RESET_STREAM_NAME)
+        .map(AirbyteStreamState::getStreamState)
+        .contains(Jsons.emptyObject());
+
+    Assertions.assertThat(emptyAirbyteSource.attemptRead())
+        .isEmpty();
+  }
+
+  @Test
+  public void testPerStream() throws Exception {
+    List<StreamDescriptor> streamDescriptors = Lists.newArrayList(
```

Comment:
should we test some the cases we discussed yesterday? e.g. a,b,c in state but streams to reset has a,b,c,d? and then a,b,c,d in state and a,b,c in streams to reset

Reply:
Added them